### PR TITLE
Add location icons to search result page

### PR DESCRIPTION
**Pull request by rwangsc18**
## Description
Add place icons for trip plan details that match with location category (visit or eatery)

## Root Cause
N/A

## Solution
Use HTML template to dynamically display location icons, new attributes are added to PlanningSolution and TimeSectionPlace to send necessary info to search result page. 

## Testing
Did you add any new test for this change, performed manual integration tests, or both?
N/A

## Final Checks
- [x] Have you removed commented code ?
- [x] Have you used gofmt to format your code ?
- [ ] You have added unit tests

### CODE REVIEW

File: planner/planner_API.go [hunk 1]
```diff
@@ -13,22 +24,18 @@ import (
...
-	"time"
 )
 
 const (
 	ServerTimeout      = time.Second * 15
 	jobQueueBufferSize = 1000
 )
 
+var placeTypeToIconCss = map[POI.PlaceCategory](string){
+	POI.PlaceCategoryEatery: "restaurant",
```
Comment (weihesdlegend) [inline planner/planner_API.go:35]: We should define a new Enum for the Icon CSS class names.

File: planner/planner_API.go [hunk 2]
```diff
@@ -255,9 +264,13 @@ func (planner *MyPlanner) Planning(ctx context.Context, planningRequest *solutio
 				EndTime:   planningRequest.Slots[pIdx].TimeSlot.Slot.End,
 				Address:   topSolution.PlaceAddresses[pIdx],
 				URL:       topSolution.PlaceURLs[pIdx],
+				PlaceIcon: placeTypeToIconCss[topSolution.PlaceTypes[pIdx]],
 			})
 		}
 		resp.Places[sIdx] = timeSectionPlaces
+		// Rui debug
```
Comment (weihesdlegend) [inline planner/planner_API.go:271]: Please remove the debugging messages.

File: planner/planner_API.go [hunk 3]
```diff
@@ -436,3 +449,14 @@ func (planner MyPlanner) SetupRouter(serverPort string) *http.Server {
 
 	return svr
 }
+
+func printSolution(solution *solution.PlanningSolution) {
```
Comment (weihesdlegend) [inline planner/planner_API.go:453]: Remove these as well.

**Review by weihesdlegend — COMMENTED** [review]
Very nice! A few comments here, and also we need to discuss when `cacheResponse.SlotSolutionCandidate` does not have the data for existing solutions in Redis.

File: iowrappers/redis_client.go [hunk 4]
```diff
@@ -8,15 +8,16 @@ import (
...
-	"github.com/go-redis/redis/v8"
-	log "github.com/sirupsen/logrus"
-	"github.com/weihesdlegend/Vacation-planner/POI"
-	"github.com/weihesdlegend/Vacation-planner/utils"
 	"net/url"
 	"strconv"
 	"strings"
 	"sync"
 	"time"
+
```
Comment (sananand007) [inline iowrappers/redis_client.go:16]: Is this change intentional?

File: iowrappers/redis_client.go [hunk 5]
```diff
@@ -321,12 +322,13 @@ func (redisClient *RedisClient) StreamsLogging(streamName string, data map[strin
...
-	PlaceAddresses []string     `json:"place_addresses"`
-	PlaceURLs      []string     `json:"place_urls"`
+	PlaceIds       []string            `json:"place_ids"`
+	Score          float64             `json:"score"`
+	PlaceNames     []string            `json:"place_names"`
+	PlaceLocations [][2]float64        `json:"place_locations"`
+	PlaceAddresses []string            `json:"place_addresses"`
+	PlaceURLs      []string            `json:"place_urls"`
+	PlaceTypes     []POI.PlaceCategory `json:"place_types"`
 }
```
Comment (sananand007) [inline iowrappers/redis_client.go:332]: If there is `1` line change, could you make sure only that is reflected in `git diff`

File: solution/create_multi_slot_solution_candidate.go [hunk 6]
```diff
@@ -1,19 +1,22 @@
...
-	IsSet          bool         `json:"is_set"`
+	PlaceNames     []string            `json:"place_names"`
+	PlaceIDS       []string            `json:"place_ids"`
+	PlaceLocations [][2]float64        `json:"place_locations"` // lat,lng
+	PlaceAddresses []string            `json:"place_addresses"`
+	PlaceURLs      []string            `json:"place_urls"`
+	PlaceTypes     []POI.PlaceCategory `json:"place_types"`
+	Score          float64             `json:"score"`
+	IsSet          bool                `json:"is_set"`
 }
```
Comment (sananand007) [inline solution/create_multi_slot_solution_candidate.go:20]: Same here, do a `git diff` so that your changes are the only changes

File: solution/generate_planning_solutions.go [hunk 7]
```diff
@@ -3,13 +3,14 @@ package solution
 import (
 	"container/heap"
 	"context"
+	"strconv"
+	"strings"
+
```
Comment (sananand007) [inline solution/generate_planning_solutions.go:8]: remove space

File: templates/search_results_layout_template.html [hunk 8]
```diff
@@ -8,6 +8,9 @@
     <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet"
           integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
     <link rel="stylesheet" href="assets/css/styles.css">
+    <!--Google Font Icon CSS-->
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
+          rel="stylesheet">
```
Comment (sananand007) [inline templates/search_results_layout_template.html:13]: 👍

**Review by sananand007 — APPROVED** [review]
[nit]s mostly

File: solution/generate_planning_solutions.go (hunk 7 above)
Comment (weihesdlegend) [reply solution/generate_planning_solutions.go:8]: I think this space is intentional from go formatter to group different dependencies. The third party ones are grouped together.

File: planner/planner_API.go (hunk 1 above)
Comment (rwangsc18) [reply planner/planner_API.go:35]: Good idea, thanks. Will do

File: solution/create_multi_slot_solution_candidate.go (hunk 6 above)
Comment (rwangsc18) [reply solution/create_multi_slot_solution_candidate.go:20]: Thanks, vscode automatically adjusts code format, which may not be ideal in this case... I will try to minimize these code changes due to formatter.

Comment (weihesdlegend) [reply solution/create_multi_slot_solution_candidate.go:20]: I think these are fine. The official Go formatter promotes grouping dependencies by the sources.

File: planner/planner_API.go (hunk 1 above)
Comment (rwangsc18) [reply planner/planner_API.go]: A new Enum is added in commit 6088153

File: planner/planner_API.go (hunk 3 above)
Comment (rwangsc18) [reply planner/planner_API.go]: function is removed in 6088153.

File: solution/generate_planning_solutions.go [hunk 9]
```diff
@@ -88,6 +89,7 @@ func GenerateSolutions(context context.Context, matcher matching.Matcher, redisC
 
 	for mdIter.HasNext() {
 		curCandidate := CreateCandidate(mdIter, placeClusters)
+		// iowrappers.Logger.Debugf("%+v\n", curCandidate)
```
Comment (weihesdlegend) [inline solution/generate_planning_solutions.go:92]: Remove commented code

File: planner/planner_API.go [hunk 10]
```diff
@@ -13,22 +24,18 @@ import (
...
-	"strings"
-	"time"
 )
 
 const (
 	ServerTimeout      = time.Second * 15
 	jobQueueBufferSize = 1000
 )
 
+var placeTypeToIcon = map[POI.PlaceCategory](POI.PlaceIcon){
```
Comment (weihesdlegend) [inline planner/planner_API.go:34]: Nit: remove the parenthesis around map values

File: planner/planner_API.go [hunk 11]
```diff
@@ -244,17 +253,20 @@ func (planner *MyPlanner) Planning(ctx context.Context, planningRequest *solutio
 
 	topSolutions := planningResponse.Solutions
 	resp.Places = make([]TimeSectionPlaces, len(topSolutions))
+
 	for sIdx, topSolution := range topSolutions {
 		timeSectionPlaces := TimeSectionPlaces{
 			Places: make([]TimeSectionPlace, 0),
 		}
 		for pIdx, placeName := range topSolution.PlaceNames {
+			// placeIcon = findPlaceIcon(topSolution.PlaceTypes, pIdx)
```
Comment (weihesdlegend) [inline planner/planner_API.go:262]: Remove commented code

**Review by weihesdlegend — APPROVED** [review]
LGTM, please remove commented code and `git commit --amend` and then git force push to the branch before merge to `master`

File: solution/create_multi_slot_solution_candidate.go [hunk 12]
```diff
@@ -1,19 +1,22 @@
...
-	PlaceAddresses []string     `json:"place_addresses"`
-	PlaceURLs      []string     `json:"place_urls"`
-	Score          float64      `json:"score"`
-	IsSet          bool         `json:"is_set"`
+	PlaceNames     []string            `json:"place_names"`
+	PlaceIDS       []string            `json:"place_ids"`
+	PlaceLocations [][2]float64        `json:"place_locations"` // lat,lng
+	PlaceAddresses []string            `json:"place_addresses"`
+	PlaceURLs      []string            `json:"place_urls"`
+	PlaceTypes     []POI.PlaceCategory `json:"place_types"`
```
Comment (weihesdlegend) [inline solution/create_multi_slot_solution_candidate.go:17]: Should be `PlaceCategories`

File: planner/planner_API.go (hunk 10 above)
Comment (weihesdlegend) [reply planner/planner_API.go:34]: Removed

File: solution/create_multi_slot_solution_candidate.go (hunk 12 above)
Comment (weihesdlegend) [reply solution/create_multi_slot_solution_candidate.go:17]: Modified
